lib/rules/no-unused-vars.js

```diff
@@ -858,10 +861,357 @@ module.exports = {
                             messageId: "unusedVar",
                             data: unusedVar.references.some(ref => ref.isWrite())
                                 ? getAssignedMessageData(unusedVar)
-                                : getDefinedMessageData(unusedVar)
+                                : getDefinedMessageData(unusedVar),
+                            suggest: [
+                                {
+                                    messageId: "removeVar",
+                                    data: {
+                                        varName: unusedVar.name
+                                    },
+                                    fix(fixer) {
+                                        const id = unusedVar.identifiers[0];
+                                        const parent = id.parent;
+                                        const tokenBefore = sourceCode.getTokenBefore(id);
+                                        const tokenAfter = sourceCode.getTokenAfter(id);
+
+                                        /**
+                                         * Gets fixes in variable declarations and function parameters
+                                         * @param {ASTNode} node parent node of identifier
+                                         * @returns {Object} fixer object
+                                         */
+                                        function fixVariables(node) {
+                                            if (node.parent.type === "VariableDeclarator") {
```

Comment:
It looks like you're using `node.parent` a lot, which requires looking up that property each time. I'd suggest saving a reference to both the parent and parent type so you don't have to keep evaluating it each time:

```js
const parent  = node.parent;
const parentType = parent.type;

Reply:
Here `node.parent` is just used in `fixVariable` function and parent is already declared as
```js
const parent = id.parent
```
is it ok if i don't replace `node.parent` for now because it will help me if i use it later also?

Reply:
I'm not sure I understand what you're saying. I see `node.parent` multiple times in the following lines, and there's just no need to keep doing the lookup.